scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/AvroFileOperations.java

```diff
@@ -110,6 +111,25 @@ Schema getSchema() {
     return schemaSupplier.get();
   }
 
+  @Override
+  public int hashCode() {
+    return Objects.hash(getSchema(), codec.getCodec(), metadata, datumFactory);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof AvroFileOperations)) {
+      return false;
+    }
+    final AvroFileOperations<ValueT> that = (AvroFileOperations<ValueT>) obj;
+    return that.getSchema().equals(this.getSchema())
+        && that.codec.getCodec().toString().equals(this.codec.getCodec().toString())
```

Comment:
Context: CodecFactory has a specialized override of toString that prints the codec type.